### PR TITLE
create and add integration tests

### PR DESCRIPTION
**Pull request by pNovikau**
#### CHANGE TYPE

 - [ ] Major
 - [x] Minor
 - [ ] Patch

#### IMPACT, DEPENDENCIES AND TEST CONSIDERATIONS
Doesn't affect current functionality

### CODE REVIEW

File: test/CodeReview.Orchestrator.SubsystemTests/Actions/ExecuteProcess.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+            return new MoveFileFromResourceAction
+            {
+                ResourceName = _resourceName,
+                FilePath = filePath
+            };
+        }
+    }
+
+    public class MoveFileFromResourceAction : IAction
```
Comment (DiamondDragon) [inline test/CodeReview.Orchestrator.SubsystemTests/Actions/ExecuteProcess.cs]: @pNovikau i would suggest to put each class into own file

File: test/CodeReview.Orchestrator.SubsystemTests/Config.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace CodeReview.Orchestrator.SubsystemTests
+{
+    public static class Config
+    {
+        public static Assembly Assembly { get; set; }
+        public static string ResourcePath { get; set; } = string.Empty;
+        public static IConfiguration Configuration { get; set; }
+
+        public static string ExePath => Configuration?["ExePath"] ?? string.Empty;
```
Comment (DiamondDragon) [inline test/CodeReview.Orchestrator.SubsystemTests/Config.cs:12]: @pNovikau `Configuration?["ExePath"]` `?` - this is something new to me. I've never seen this kind of usage before =). Nice!

File: test/CodeReview.Orchestrator.SubsystemTests/Tests/RunTests.cs [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        [Fact]
+        public void When_All_Ok_Should_Return_Ok()
+        {
+            const string srcResourceName = "Run.src.txt";
+            const string srcOutputName = nameof(RunTests) + "src.txt";
+            
+            var srcFilePath = Path.Combine("src", srcOutputName);
+            var importFilePath = Path.Combine("imports", srcOutputName);
+            
+            var srcVolumeId = $"src-{Guid.NewGuid().ToString()}";
```
Comment (DiamondDragon) [inline test/CodeReview.Orchestrator.SubsystemTests/Tests/RunTests.cs]: @pNovikau `.ToString()` is redundant here.

File: test/CodeReview.Orchestrator.SubsystemTests/Tests/RunTests.cs [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+                    .HasPerformed<StubDockerCreateVolume>(m => m.WithVolumePrefix("src").WithResponseVolumeId(srcVolumeId))
+                    .HasPerformed<StubDockerCreateVolume>(m => m.WithVolumePrefix("art").WithResponseVolumeId(artVolumeId))
+                    .HasPerformed<StubDockerCreateVolume>(m => m.WithVolumePrefix("imp").WithResponseVolumeId(impVolumeId))
+                    
+                    .HasPerformed<StubDockerImportDataActivity>(m => m.WithImage(AlpineImage).WithSourceId(impVolumeId))
+                    .HasPerformed<StubDockerImportSourceActivity>(m => m.WithImage(AlpineImage).WithSourceId(srcVolumeId))
+                
+                    .HasPerformed<StubDockerContainerGitActivity>(m => m
+                        .WithEnv("GIT_REPOSITORY_URL=https://github.com/GodelTech/CodeReview.Orchestrator.git")
+                        .WithEnv("GIT_BRANCH=main")
```
Comment (DiamondDragon) [inline test/CodeReview.Orchestrator.SubsystemTests/Tests/RunTests.cs]: I think `.WithEnv(string variableName, string value)` would look better.

**Review by DiamondDragon — DISMISSED** [review]
In general PR looks good. Few minor changes were added.
I think we can merge this PR and we need to create second PR which is going contain pipeline changes and some description how to run these tests locally.

File: test/CodeReview.Orchestrator.SubsystemTests/Config.cs (hunk 2 above)
Comment (pNovikau) [reply test/CodeReview.Orchestrator.SubsystemTests/Config.cs:12]: C# 6.0 magic :mage_man:

**Review by DiamondDragon — APPROVED** [review]
Looks good!

**Pull request merged**